frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -377,12 +402,12 @@ fn decl_all_modules<'a>(
 }
 
 fn decl_module_to_index<'a>(
-	module_declarations: impl Iterator<Item = &'a ModuleDeclaration>,
-	num_modules: usize,
+	module_declarations: &Vec<ModuleDeclaration>,
```

Comment:
Why not an iterator anymore?

Reply:
because now quote! makes use of two iterators (though we could factorize to make use of only one)

Reply:
Okay, but than at least you should pass a `slice`

Reply:
yes I rewrote with slice or back to iterator where it was finally not needed, I was negligent on internal functions better to have proper types indeed.